pkg/artifactory/resource/configuration/resource_artifactory_property_set.go

```diff
@@ -0,0 +1,290 @@
+									Required:         true,
+									Description:      `Predefined property name.`,
+									ValidateDiagFunc: validator.StringIsNotEmpty,
+								},
+								"default_value": {
+									Type:        schema.TypeBool,
+									Required:    true,
+									Description: `Whether the value is selected by default in the UI.`,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	var resourcePropertySetRead = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		propertySetConfigs := &PropertySets{}
+		// Unpacking HCL to compare the names of the property sets with the XML data we will get from the API
+		unpackedPropertySet := unpackPropertySet(d)
+
+		_, err := m.(*resty.Client).R().SetResult(&propertySetConfigs).Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+		var matchedPropertySet = PropertySet{}
+
+		matchedPropertySet = PropertySet{}
+		for _, iterPropertySet := range propertySetConfigs.PropertySets {
+			if iterPropertySet.Name == unpackedPropertySet.Name {
+				matchedPropertySet = iterPropertySet
+				break
+			}
+		}
+
+		pkr := packer.Universal(
+			predicate.All(
+				predicate.SchemaHasKey(propertySetsSchema),
+			),
+		)
+
+		return diag.FromErr(pkr(&matchedPropertySet, d))
+	}
+
+	var parsePredefinedValues = func(values []PredefinedValue) map[string]interface{} {
+		parsedPredefinedValues := map[string]interface{}{}
+		for _, value := range values {
+			parsedPredefinedValues[value.Name] = map[string]interface{}{
+				"defaultValue": value.DefaultValue,
+			}
+		}
+		return parsedPredefinedValues
+	}
+
+	var parseProperties = func(properties []Property) map[string]interface{} {
+		parsedProperties := map[string]interface{}{}
+
+		for _, property := range properties {
+			parsedProperties[property.Name] = map[string]interface{}{
+				"predefinedValues":       parsePredefinedValues(property.PredefinedValues),
+				"closedPredefinedValues": property.ClosedPredefinedValue,
+				"multipleChoice":         property.MultipleChoice,
+			}
+		}
+		return parsedProperties
+	}
+
+	var resourcePropertySetsUpdate = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		unpackedPropertySet := unpackPropertySet(d)
+
+		///* EXPLANATION FOR BELOW CONSTRUCTION USAGE.
+		//There is a difference in xml structure usage between GET and PATCH calls of API: /artifactory/api/system/configuration.
+		//GET call structure has "propertySets -> propertySet -> Array of property sets".
+		//PATCH call structure has "propertySets -> propertySet (dynamic sting). Property name and predefinedValues names are also dynamic strings".
+		//Following nested map of string structs are constructed to match the usage of PATCH call with the consideration of dynamic strings.
+		//*/
+		var constructBody = map[string]map[string]map[string]interface{}{
```

Comment:
`constructBody` is a verb and this should be a noun. Let's called this `body`. 😄

Reply:
ok